### PR TITLE
[TT-4709] Cleanup OAS authentication code

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

File: apidef/oas/security_test.go [hunk 1]
```diff
@@ -512,11 +512,11 @@ func TestOAS_GoPlugin(t *testing.T) {
 	}
 
 	var api apidef.APIDefinition
-	oas.extractGoPluginTo(&api)
+	oas.getTykAuthentication().ExtractTo(&api)
 
 	var convertedOAS OAS
 	convertedOAS.SetTykExtension(&XTykAPIGateway{Server: Server{Authentication: &Authentication{}}})
-	convertedOAS.fillGoPlugin(api)
+	convertedOAS.getTykAuthentication().Fill(api)
```
Comment (titpetric) [inline apidef/oas/security_test.go:519]: Do we need nil checks for getTykAuthentication here, I see a nil check higher up

**Review by titpetric — APPROVED** [review]

File: apidef/oas/security_test.go (hunk 1 above)
Comment (furkansenharputlu) [reply apidef/oas/security_test.go:519]: In the line above, it is initialized: 
```
convertedOAS.SetTykExtension(&XTykAPIGateway{Server: Server{Authentication: &Authentication{}}})
```

File: apidef/oas/security.go [hunk 2]
```diff
@@ -21,6 +19,19 @@ const (
 	cookie = "cookie"
 )
 
+type Token struct {
+	// Enabled enables the token based authentication mode.
+	// Old API Definition: `api_id`
```
Comment (jeffy-mathew) [inline apidef/oas/security.go:24]: can we correct this?

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3963/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2052521074)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3963/merge
Commit: 759aaf6d1885260d56a67b8d397d1461cf0daa31 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2052537524)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3963/merge
Commit: 8b47513789e900588f3859ef71b5949067ce339d 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2052540412)

File: apidef/oas/security.go [hunk 3]
```diff
@@ -21,6 +19,19 @@ const (
...
+	Enabled     bool `bson:"enabled" json:"enabled"` // required
+	AuthSources `bson:",inline" json:",inline"`
+	// EnableClientCertificate allows to create dynamic keys based on certificates.
+	// Old API Definition: `auth_configs["authToken"].use_certificate`
+	EnableClientCertificate bool `bson:"enableClientCertificate,omitempty" json:"enableClientCertificate,omitempty"`
+	//
+	// Old API Definition:
```
Comment (jeffy-mathew) [inline apidef/oas/security.go:31]: do we need this?

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 3e5d3e74a45e391435fded9e1a0a234a64adee39 Cleanup OAS authentication code (#3963)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2052655050)

**Pull request merged**